jetty-core/jetty-deploy/src/test/java/org/eclipse/jetty/deploy/DeploymentScannerCoreWebappTest.java

```diff
@@ -0,0 +1,315 @@
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@ExtendWith(WorkDirExtension.class)
+public class DeploymentScannerCoreWebappTest extends AbstractCleanEnvironmentTest
+{
+    public WorkDir workDir;
+    private Server server;
+
+    public void startServer(DeploymentManager deploymentManager) throws Exception
+    {
+        server = new Server();
+        ServerConnector connector = new ServerConnector(server);
+        connector.setPort(0);
+        server.addConnector(connector);
+
+        ContextHandlerCollection contexts = new ContextHandlerCollection();
+        server.setHandler(contexts);
+
+        deploymentManager.setContexts(contexts);
+        server.addBean(deploymentManager);
+
+        server.start();
+    }
+
+    @AfterEach
+    public void stopServer()
+    {
+        LifeCycle.stop(server);
+    }
+
+    @Test
+    public void testExampleCoreDir() throws Exception
+    {
+        Path baseDir = workDir.getEmptyPathDir();
+
+        Path webapps = baseDir.resolve("webapps");
+        FS.ensureDirExists(webapps);
+
+        Path demoDir = webapps.resolve("demo.d");
+        FS.ensureDirExists(demoDir);
+
+        Path srcZip = MavenPaths.targetDir().resolve("core-webapps/jetty-test-core-example-webapp.zip");
+        Assertions.assertTrue(Files.exists(srcZip), "Src Zip should exist: " + srcZip);
+        unpack(srcZip, demoDir);
+
+        // ensure that demo jar isn't in our test/server classpath.
+        // it should only exist in the jar file on disk.
+        assertThrows(ClassNotFoundException.class, () -> Class.forName("org.example.ExampleHandler"));
+
+        Path demoXml = webapps.resolve("demo.xml");
+        String demoXmlStr = """
+            <?xml version="1.0"?>
+            <!DOCTYPE Configure PUBLIC "-//Jetty//Configure//EN" "https://jetty.org/configure.dtd">
+            <Configure class="org.eclipse.jetty.server.handler.CoreContextHandler">
+              <Set name="contextPath">/demo</Set>
+              <Set name="handler">
+                <New class="org.example.ExampleHandler" />
+              </Set>
+            </Configure>
+            """;
+        Files.writeString(demoXml, demoXmlStr);
+
+        DeploymentManager deploymentManager = new DeploymentManager();
+        DeploymentScanner defaultProvider = new DeploymentScanner(server, deploymentManager);
+        defaultProvider.addMonitoredDirectory(webapps);
+        DeploymentScanner.EnvironmentConfig coreConfig = defaultProvider.configureEnvironment("core");
+        coreConfig.setContextHandlerClass(CoreContextHandler.class.getName());
+        deploymentManager.addBean(defaultProvider);
+        startServer(deploymentManager);
```

Comment:
This is not the order/structure that the modules create.  Currently the modules/xml structure would be represented by the code below (but I'm not sure we want it that way):

```suggestion
        server = new Server();
        ContextHandlerCollection contexts = new ContextHandlerCollection();
        server.setHandler(contexts);
        ServerConnector connector = new ServerConnector(server);
        connector.setPort(0);
        server.addConnector(connector);

        DeploymentManager deploymentManager = new DeploymentManager();
        deploymentManager.setContexts(contexts);
        server.addBean(deploymentManager);

        DeploymentScanner deploymentScanner = new DeploymentScanner(server, deploymentManager);
        deploymentScanner.addMonitoredDirectory(webapps);
        DeploymentScanner.EnvironmentConfig coreConfig = deploymentScanner.configureEnvironment("core");
        coreConfig.setContextHandlerClass(CoreContextHandler.class.getName());

        server.start();
        deploymentScanner.start();
```